frontend/src/components/pages/main/MainPage.tsx

```diff
@@ -1,27 +1,85 @@
+import { useEffect, useState } from "react";
+
 import { css } from "@emotion/react";
 
+import { Tag } from "@type/domain/travelogue";
+
+import useGetTags from "@queries/useGetTags";
 import useInfiniteTravelogues from "@queries/useInfiniteTravelogues";
 
+import { Chip, Text } from "@components/common";
 import FloatingButton from "@components/common/FloatingButton/FloatingButton";
 import TravelogueCard from "@components/pages/main/TravelogueCard/TravelogueCard";
 
+import { useDragScroll } from "@hooks/useDragScroll";
 import useIntersectionObserver from "@hooks/useIntersectionObserver";
 
 import * as S from "./MainPage.styled";
 import TravelogueCardSkeleton from "./TravelogueCard/skeleton/TravelogueCardSkeleton";
 
 const MainPage = () => {
   const SKELETON_COUNT = 5;
-  const { travelogues, status, fetchNextPage } = useInfiniteTravelogues();
+  const MAX_TAGS_COUNT = 3;
+
+  const { data: tags } = useGetTags();
+
+  const [sortedTags, setSortedTags] = useState<Tag[]>([]);
+  const [selectedTagIDs, setSelectedTagIDs] = useState<number[]>([]);
+
+  useEffect(() => {
+    if (tags) {
+      setSortedTags(tags);
+    }
+  }, [tags]);
+
+  const { travelogues, status, fetchNextPage } = useInfiniteTravelogues(selectedTagIDs);
   const { lastElementRef } = useIntersectionObserver(fetchNextPage);
 
+  const handleClickChip = (id: number) => {
+    setSelectedTagIDs((prevSelectedTagIDs) => {
+      const newSelectedTagIDs = prevSelectedTagIDs.includes(id)
+        ? prevSelectedTagIDs.filter((selectedTagID) => selectedTagID !== id)
+        : [...prevSelectedTagIDs, id];
+
+      if (newSelectedTagIDs.length > MAX_TAGS_COUNT) return prevSelectedTagIDs;
+
+      setSortedTags((prevSortedTags) => {
+        const selected = prevSortedTags.filter((tag) => newSelectedTagIDs.includes(tag.id));
+        const unselected = prevSortedTags.filter((tag) => !newSelectedTagIDs.includes(tag.id));
+        return [...selected, ...unselected];
+      });
+
+      return newSelectedTagIDs;
+    });
+  };
```

Comment:
```suggestion
  const createSortedTags = () => {
    if (!tags) return [];

    const selected = tags.filter((tag) => selectedTagIDs.includes(tag.id));
    const unselected = tags.filter((tag) => !selectedTagIDs.includes(tag.id));

    return [...selected, ...unselected];
  };

  const { travelogues, status, fetchNextPage } = useInfiniteTravelogues(selectedTagIDs);
  const { lastElementRef } = useIntersectionObserver(fetchNextPage);

  const handleClickChip = (id: number) => {
    setSelectedTagIDs((prevSelectedTagIDs) => {
      const newSelectedTagIDs = prevSelectedTagIDs.includes(id)
        ? prevSelectedTagIDs.filter((selectedTagID) => selectedTagID !== id)
        : [...prevSelectedTagIDs, id];

      if (newSelectedTagIDs.length > MAX_TAGS_COUNT) return prevSelectedTagIDs;

      return newSelectedTagIDs;
    });
  };
```

이렇게 useEffect와 별도의 state(sortedTags)를 두지 않고 계산으로 처리해볼 수 있을거 같아요!

**장점**
1. setSelectedTagIDs 내에서 setSortedTags를 호출하지 않아 사이드 이펙트를 방지할 수 있다.
2. 불필요한 useEffect를 줄일 수 있다. (effect는 외부 동기화 목적으로 사용하는 hook이기 때문!)

Reply:
처음에 코드를 보고 정상 작동하나라고 궁금했는데 리액트 동작 원리를 잘 이해한 지니 덕분에 이렇게 좋은 코드를 볼 수 있네요 감탄했습니다 좋은 코드 감사합니다